lib/ruboty/slack_socket_mode/interactive_message.rb

```diff
@@ -46,6 +46,37 @@ def delete
 
         robot.delete_interactive(response_url)
       end
+
+      def update(text: nil, blocks: nil)
+        response_url = @payload['response_url']
+        unless response_url
+          Ruboty.logger.warn("#{self.class.name}: Cannot update message. This message does not contain response_url in payload.")
+          return
+        end
+        robot.update_interactive(response_url, text, blocks)
+      end
+
+      def state(block_id:, action_id:)
+        state = @payload.dig('state', 'values', block_id, action_id)
+        unless state
+          Ruboty.logger.warn("#{self.class.name}: Cannot get state. Block_id or action_id is not found")
+          return
+        end
+        return case state['type']
+        when 'static_select' then state.dig('selected_option', 'value')
+        when 'multi_static_select' then state.dig('selected_options').map { |selected_option| selected_option['value'] }
+        when 'multi_conversations_select' then state.dig('selected_conversations')
+        when 'checkboxes' then state.dig('selected_options').map { |selected_option| selected_option['value'] }
+        when 'radio_buttons' then state.dig('selected_option', 'value')
+        when 'timepicker' then state.dig('selected_time')
+        when 'users_select' then state.dig('selected_user')
+        when 'multi_users_select' then state.dig('selected_users')
+        when 'datepicker' then state.dig('selected_date')
+        when 'plain_text_input' then state.dig('value')
+        else
+          Ruboty.logger.warn("#{self.class.name}: Cannot get state. This is unsupported type.")
```

Comment:
[imo] `warn` メソッドの返値が `true` のようで、このメソッドの最終的な返値も `true` になってしまうので、一応安全のため・・ `nil` を返すようにしておくと良さそうです。

```suggestion
          Ruboty.logger.warn("#{self.class.name}: Cannot get state. This is unsupported type.")
          return nil
```